### ISSUE TITLE
Spoiler-free fictional universe wiki platform

### ISSUE THREAD

**Issue by ItsNickBarry** [open-source-ideas/open-source-ideas#198]
## Project description

When engaging with a literary or cinematic fictional universe over a long period of time, it's easy to forget certain details that become important later.  It's not as easy to look up those details on the internet without learning too much: wikis, the most reliable sources of such esoteric information, invariably contain spoilers.

I propose a wiki platform which allows users to filter data based on their unique positions within a narrative.  The mechanism behind this would be nothing more than source citations, a feature that is already supported by most wikis, and one that would be well treated as an epistemological standard.

This data-based approach to wiki organization comes at the expense of the narrative prose usually employed by wikis.  Instead, the main content of each article would probably be in list form, while some data would be best displayed on a timeline.  Considering the writing style often encountered on public wikis, this may be considered an advantage.

### Example

Imagine that a user is reading the Book of Genesis.

Our reader arrives at verse 20: 
> Two of every kind of bird, of every kind of animal and of every kind of creature that moves along the ground will come to you to be kept alive.

"But wait," our reader exclaims.  "How will they fit?  Surely the boat is not enough cubits long!  Yet memories are fickle, and I do not recall the specifications."

Our reader then accesses the wiki for the Book of Genesis, navigates to the "Noah's Ark" page, and sets the filter to exclude information which appears later than verse 20 of chapter 6.  The wiki returns the relevant data, including a description of verse 15 of chapter 6:
> This is how you are to build it: The ark is to be three hundred cubits long, fifty cubits wide and thirty cubits high. 

"Indeed, I suppose that is enough cubits," our reader thinks, reassured.

Meanwhile the wiki does not display the spoiler contained in verse 11 of chapter 8:
> When the dove returned to him in the evening, there in its beak was a freshly plucked olive leaf! Then Noah knew that the water had receded from the earth.

Thus, our reader is able to encounter information in the order that the author intended.

## Relevant Technology

* standard web stack

### Complexity

- [ ] Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- [x] Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- [ ] Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- [ ] Little work - A couple of days
- [ ] Medium work - A week or two
- [x] Much work - The project will take more than a couple of weeks and serious planning is required

### Categories

- [ ] Mobile app
- [ ] IoT
- [x] Web app
- [ ] Frontend/UI
- [ ] AI/ML
- [ ] APIs/Backend
- [ ] Voice Assistant
- [ ] Developer Tooling
- [ ] Extension/Plugin/Add-On
- [ ] Design/UX
- [ ] AR/VR
- [ ] Bots
- [ ] Security
- [ ] Blockchain
- [ ] Futuristic Tech/Something Unique

**Comment by remram44** [comment]
This seems like a good idea! There are plenty of wiki platforms and the like for fictional universe, but you'll get spoiled every time. The data on when things happen/are discovered is usually there, it would just need some software support to not show you those spoilders :sweat_smile:

**Comment by jusdepatate** [comment]
So you just need a wiki with a good moderation team ready to remove any spoiler ?
Then the only difficulty is to get an active&good mod team not to make the wiki there are plenty of wiki software easy to host if i understand correctly

**Comment by ItsNickBarry** [comment]
@jusdepatate No, spoilers are only spoilers when not presented in the right context, and this context must be built into the data model.  When adding information to the wiki, a user would have to intentionally cite a source  incorrectly for spoilers to appear.

Also, because these would not be written in prose, it might be okay if certain citations remain invisible until verified.  (In the case of prose, race conditions in edits and verifications would sometimes result in un-mergable document histories [like git merge conflicts], but not in this case).

**Comment by Nitemice** [comment]
An example of a website that does something like this is [this Veronica Mars fansite](https://marsinvestigations.net/).
It has a dropdown menu in the top-right corner which allows you to select the last episode you've seen, and limits what appears on the site based on that.

From playing around with it a little, it seems like basically every chunk of info is tagged by what episode it's revealed in, and then it just filters all chunks from episodes after the one you are up to.

It would be a lot of work to tag every little scrap of info like that, but depending on the fanbase, there'd probably be someone interested in doing it.

**Comment by TheOtterlord** [comment]
> It would be a lot of work to tag every little scrap of info like that, but depending on the fanbase, there'd probably be someone interested in doing it.

@Nitemice There will always be one fan 😆 

Jokes aside, this is a really interesting way of doing it. It does not have to be as indepth as the example above either. You could tag either pages, sections or paragraphs, the problem is keeping it moderated. You'd probably end up with a system that could be overthrown by a few trolls, or so strict that every edit needs to be checked by a mod unless the person is a trusted community member (role?).

**Comment by Nitemice** [comment]
> You'd probably end up with a system that could be overthrown by a few trolls, or so strict that every edit needs to be checked by a mod unless the person is a trusted community member (role?).

At that point, it becomes less of a programming/software problem, and more of a management/moderation problem. And even then, I think it would be highly dependant on the fandom in question. If it's something high-profile, or where even a small spoiler could really ruin everything, then that would be bad. But even then, it's not really that different from the typical spam any community/wiki is likely to attract.
The only way I can see this being worse, is that it explicitly indicates to a random, unknowledgeable troll what information is likely to ruin a typical fans day. And I really don't see any way of fixing that. You can't lock down the spoilers.

That's not to say that you couldn't or shouldn't try to build a solution into the software. I just don't see one.
